meta-balena-common/recipes-support/hostapp-update-hooks/files/99-resin-uboot

```diff
@@ -22,7 +22,14 @@ new_part_idx=$(cat "/sys/class/block/$blockdev/partition")
 new_part_idx=$(printf "%x" $new_part_idx)
 
 printf "[INFO] Switching uboot root partition index to %s..." "$new_part_idx..."
-echo "resin_root_part=$new_part_idx" > /mnt/boot/resinOS_uEnv.txt.new
+
+touch /mnt/boot/resinOS_uEnv.txt
+grep -v "resin_root_part" /mnt/boot/resinOS_uEnv.txt > /mnt/boot/resinOS_uEnv.txt.new
+uenv=$(cat /mnt/boot/resinOS_uEnv.txt.new)
+if [ ! -z "$uenv" ]; then
+	echo "$uenv" | grep -v "upgrade_available"  > /mnt/boot/resinOS_uEnv.txt.new
+fi
+echo "resin_root_part=$new_part_idx" >> /mnt/boot/resinOS_uEnv.txt.new
```

Comment:
My first impression was that I'd rather not mangle so much with the u-boot environment at HUP. Previously it was very clean, just creating a new file with a couple of variables.

Couple of questions:
1. How is the supervisor appending the extra cmd line arguments? Is it modifying resinOS_uEnv directly or keeping them separately in supervisor.conf?
2. Did you consider any alternative to the above?

Reply:
Removed this part as discussed, added a different environment file